### ISSUE TITLE
WebStorm 2017.3 EAP 173.2696.21

### ISSUE THREAD

**Issue by sibelius** [jimkyndemeyer/js-graphql-intellij-plugin#104]
this should work with node 8?

```java
Unable to start JS GraphQL Language Service using Node.js with commandline /Users/sibelius/.nvm/versions/node/v8.4.0/bin/node [/Users/sibelius/Library/Application Support/WebStorm2017.3/js-graphql-language-service.dist.js, --port=63244]
java.lang.Throwable: Unable to start JS GraphQL Language Service using Node.js with commandline /Users/sibelius/.nvm/versions/node/v8.4.0/bin/node [/Users/sibelius/Library/Application Support/WebStorm2017.3/js-graphql-language-service.dist.js, --port=63244]
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:136)
	at com.intellij.lang.jsgraphql.languageservice.JSGraphQLNodeLanguageServiceInstance.createProcessHandler(JSGraphQLNodeLanguageServiceInstance.java:143)
	at com.intellij.lang.jsgraphql.languageservice.JSGraphQLNodeLanguageServiceInstance.<init>(JSGraphQLNodeLanguageServiceInstance.java:83)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at com.intellij.lang.jsgraphql.languageservice.JSGraphQLNodeLanguageServiceClient.getJSGraphQLNodeLanguageServiceInstance(JSGraphQLNodeLanguageServiceClient.java:118)
	at com.intellij.lang.jsgraphql.languageservice.JSGraphQLNodeLanguageServiceClient.executeRequest(JSGraphQLNodeLanguageServiceClient.java:71)
	at com.intellij.lang.jsgraphql.languageservice.JSGraphQLNodeLanguageServiceClient.executeRequest(JSGraphQLNodeLanguageServiceClient.java:66)
	at com.intellij.lang.jsgraphql.languageservice.JSGraphQLNodeLanguageServiceClient.getSchemaWithVersion(JSGraphQLNodeLanguageServiceClient.java:62)
	at com.intellij.lang.jsgraphql.schema.ide.project.JSGraphQLSchemaLanguageProjectService.getOrCreateSchemaFileElements(JSGraphQLSchemaLanguageProjectService.java:626)
	at com.intellij.lang.jsgraphql.schema.ide.project.JSGraphQLSchemaLanguageProjectService.getSchemaFile(JSGraphQLSchemaLanguageProjectService.java:115)
	at com.intellij.lang.jsgraphql.schema.ide.project.JSGraphQLSchemaDirectoryNode.getChildren(JSGraphQLSchemaDirectoryNode.java:53)
	at com.intellij.ide.util.treeView.AbstractTreeStructureBase.getChildElements(AbstractTreeStructureBase.java:47)
	at com.intellij.ide.util.treeView.AbstractTreeUi$31.perform(AbstractTreeUi.java:1619)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:36)
	at com.intellij.ide.util.treeView.AbstractTreeUi.execute(AbstractTreeUi.java:1856)
	at com.intellij.ide.util.treeView.AbstractTreeUi.getChildrenFor(AbstractTreeUi.java:1616)
	at com.intellij.ide.util.treeView.AbstractTreeUi.access$4600(AbstractTreeUi.java:64)
	at com.intellij.ide.util.treeView.AbstractTreeUi$30.perform(AbstractTreeUi.java:1542)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:36)
	at com.intellij.ide.util.treeView.AbstractTreeUi.execute(AbstractTreeUi.java:1856)
	at com.intellij.ide.util.treeView.AbstractTreeUi.processUnbuilt(AbstractTreeUi.java:1524)
	at com.intellij.ide.util.treeView.AbstractTreeUi.doUpdateChildren(AbstractTreeUi.java:1250)
	at com.intellij.ide.util.treeView.AbstractTreeUi.access$3000(AbstractTreeUi.java:64)
	at com.intellij.ide.util.treeView.AbstractTreeUi$25$1.perform(AbstractTreeUi.java:1173)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:36)
	at com.intellij.ide.util.treeView.AbstractTreeUi.execute(AbstractTreeUi.java:1856)
	at com.intellij.ide.util.treeView.AbstractTreeUi.access$2600(AbstractTreeUi.java:64)
	at com.intellij.ide.util.treeView.AbstractTreeUi$25.perform(AbstractTreeUi.java:1170)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:36)
	at com.intellij.openapi.util.ActionCallback$ExecutedExecutionCallback.doWhenExecuted(ActionCallback.java:177)
	at com.intellij.openapi.util.ActionCallback.doWhenDone(ActionCallback.java:111)
	at com.intellij.ide.util.treeView.AbstractTreeUi.updateNodeChildren(AbstractTreeUi.java:1165)
	at com.intellij.ide.util.treeView.AbstractTreeUi.updateNodesToInsert(AbstractTreeUi.java:1668)
	at com.intellij.ide.util.treeView.AbstractTreeUi.access$4300(AbstractTreeUi.java:64)
	at com.intellij.ide.util.treeView.AbstractTreeUi$29.lambda$perform$0(AbstractTreeUi.java:1395)
	at com.intellij.openapi.util.AsyncResult.lambda$doWhenDone$1(AsyncResult.java:77)
	at com.intellij.openapi.util.ExecutionCallback.doWhenExecuted(ExecutionCallback.java:100)
	at com.intellij.openapi.util.ActionCallback.doWhenDone(ActionCallback.java:111)
	at com.intellij.openapi.util.AsyncResult.doWhenDone(AsyncResult.java:77)
	at com.intellij.ide.util.treeView.AbstractTreeUi$29.perform(AbstractTreeUi.java:1393)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:36)
	at com.intellij.ide.util.treeView.TreeRunnable$TreeConsumer.consume(TreeRunnable.java:50)
	at org.jetbrains.concurrency.AsyncPromise.setResult(AsyncPromise.kt:152)
	at org.jetbrains.concurrency.DonePromise.notify(DonePromise.java:101)
	at com.intellij.ide.util.treeView.AbstractTreeUi$33.perform(AbstractTreeUi.java:1801)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:36)
	at com.intellij.ide.util.treeView.AbstractTreeUi.executeYieldingRequest(AbstractTreeUi.java:2090)
	at com.intellij.ide.util.treeView.AbstractTreeUi.access$5900(AbstractTreeUi.java:64)
	at com.intellij.ide.util.treeView.AbstractTreeUi$36$1.perform(AbstractTreeUi.java:2002)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:36)
	at com.intellij.ide.util.treeView.AbstractTreeBuilder.runOnYeildingDone(AbstractTreeBuilder.java:410)
	at com.intellij.ide.util.treeView.AbstractTreeUi.runOnYieldingDone(AbstractTreeUi.java:2185)
	at com.intellij.ide.util.treeView.AbstractTreeUi$36.perform(AbstractTreeUi.java:1997)
	at com.intellij.ide.util.treeView.TreeRunnable.run(TreeRunnable.java:36)
	at com.intellij.ide.util.treeView.AbstractTreeBuilder.lambda$new$0(AbstractTreeBuilder.java:49)
	at com.intellij.util.containers.TransferToEDTQueue.processNext(TransferToEDTQueue.java:102)
	at com.intellij.util.containers.TransferToEDTQueue.access$300(TransferToEDTQueue.java:37)
	at com.intellij.util.containers.TransferToEDTQueue$1.run(TransferToEDTQueue.java:57)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:822)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:650)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:366)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by jimkyndemeyer** [comment]
Hi sibelius.

I haven't heard of any issues with Node 8. I think this relates to #103.

The EAP no longer ships with Apache Commons (https://youtrack.jetbrains.com/issue/WEB-28968), and this plugin uses the Apache HTTP Client to connect to the language service.

**Comment by jimkyndemeyer** [comment]
Fixed with 1.6.1, pending JetBrains approval.

**Comment by jimkyndemeyer** [comment]
Fixed with PR #105

**Comment by sibelius** [comment]
awesome work, keep going
